cardshifter-server/src/test/java/com/cardshifter/server/main/ServerConnectionTest.java

```diff
@@ -252,5 +252,16 @@ public void testUsersQueryNotLoggedIn() throws IOException, InterruptedException
 		// There shouldn't be a UserStatusMessage for client2
 		assertNoMessage(client1);
 	}
+
+	@Test(timeout = 5000)
+	public void testNoOfflineMessageIfNotLoggedIn() throws IOException, InterruptedException {
+		System.out.println("testNoOfflineMessageIfNotLoggedIn");
+		TestClient client2 = createTestClient();
+		client2.disconnect();
+
+		// Avoid race condition between UserStatusMessage and ServerStatusMessage sent by assertNoMessage
+		Thread.sleep(1000);
```

Comment:
Thread.sleep does not seem like the proper way to handle this. Can you explain the problem in more detail? Then we can figure out another solution. What are the messages being sent if this is not here? What is the race condition exactly?